src/modules/prepareSpecificSettingToSave.js

```diff
@@ -0,0 +1,133 @@
+import { populateSettingObject } from './dataset/populateSettingObject'
+import { DATA_SET, DEFAULT, SPECIFIC } from '../constants/data-set-settings'
+import {
+    PROGRAM,
+    specificSettingsDefault,
+    WITH_REGISTRATION,
+    WITHOUT_REGISTRATION,
+} from '../constants/program-settings'
+import { populateProgramObject } from './programs/populateProgramObject'
+
+/**
+ * Prepare specific settings to be save
+ */
+export const prepareSpecificSettingsToSave = ({
+    settingType,
+    states,
+    specificSettings,
+    specificSettingToChange,
+    specificSettingsRows,
+    specificSettingsNameList,
+    settingsCompleteList,
+}) => {
+    const specificNameKey = states.specificSetting.name
+    const objData = specificSettings
+
+    const settingNameFilter = settingsCompleteList.filter(
+        option => option.id === specificNameKey
+    )
+
+    if (settingNameFilter.length > 0) {
```

Comment:
This would be a little cleaner using `find` instead of `filter`

```js
const settingNameFilter = settingsCompleteList.find(option => option.id === specificNameKey)
if (settingNameFilter) {
   // ... use `settingNameFilter` instead of `settingNameFilter[0]`
}
```